Client/Frontend/Settings/AdblockDebugMenuTableViewController.swift

```diff
@@ -37,82 +51,128 @@ class AdblockDebugMenuTableViewController: TableViewController {
+                
+                var text = $0 + ".json"
+                
+                // Rules with count = 1 don't have to be shown, they are static rules for cookie control
+                // tracking protection and httpse.
+                if json.count > 1 {
+                    text += ": \(json.count) rules"
+                }
+                
+                let hashText = "sha1: \(jsonData.sha1.hexEncodedString)"
+                rows.append(.init(text: text, detailText: hashText, cellClass: ShrinkingSubtitleCell.self))
             }
             
-            // Search in application support directory otherwise
-            let folderName = AdblockResourceDownloader.folderName
-            guard let folderUrl = fm.getOrCreateFolder(name: folderName) else { return nil }
-            
-            let fileUrl = folderUrl.appendingPathComponent(name + ".json")
-            return fm.contents(atPath: fileUrl.path)
+            if $0 == "block-ads", let bundlePath = Bundle.main.path(forResource: "ABPFilterParserData",
+                                                                    ofType: "dat"),
+                let data = fm.contents(atPath: bundlePath) {
+                let hashText = "sha1: \(data.sha1.hexEncodedString)"
+                rows.append(.init(text: "ABPFilterParserData.dat", detailText: hashText, cellClass: ShrinkingSubtitleCell.self))
+            }
         }
+    
+        section.rows = rows
+        return section
+    }
+    
+    private func downloadedListsSection(names: [String]) -> Section {
+        var section = Section(header: "Downloaded lists",
+                              footer: "Lists downloaded from the internet at app launch.")
+        
+        var rows = [Row]()
         
         func getEtag(name: String) -> String? {
             let fm = FileManager.default
             
             guard let folderUrl = fm.getOrCreateFolder(name: AdblockResourceDownloader.folderName) else {
                 return nil
             }
-            let etagUrl = folderUrl.appendingPathComponent(name + ".json.etag")
+            let etagUrl = folderUrl.appendingPathComponent(name + ".etag")
             guard let data = fm.contents(atPath: etagUrl.path) else { return nil }
             return String(data: data, encoding: .utf8)
         }
         
-        store.getAvailableContentRuleListIdentifiers { list in
-            list?.forEach {
-                var row = Row(text: $0, cellClass: ShrinkingSubtitleCell.self)
+        func getLastmodified(name: String) -> String? {
+            let fm = FileManager.default
+            
+            guard let folderUrl = fm.getOrCreateFolder(name: AdblockResourceDownloader.folderName) else {
+                return nil
+            }
+            let etagUrl = folderUrl.appendingPathComponent(name + ".lastmodified")
```

Comment:
I doubt this works when I download a file from the server, no?

Reply:
Sure it does, I've been using it for Brave Today :) 

https://github.com/brave/brave-ios/blob/6bd51840421f3ecccf0957b83372874ba0a7a73c/Client/Frontend/Brave%20Today/Composer/FeedDataSource.swift#L202-L211

Reply:
we talked in DMs, and we have to use file-modified header because we care about time when the file was modified on the server, not locally on disk

going to disable persisting it for public builds too